dataprofiler/profilers/profile_builder.py

```diff
@@ -1372,17 +1367,26 @@ def _get_correlation(self, clean_samples):
         :type clean_samples: dict()
         """
         columns = self.options.correlation.columns
+        clean_column_ids = []
         if columns is None:
-            for col in self._profile:
+            for col_id, col in enumerate(self._profile.keys()):
                 if (self._profile[col].profile['data_type'] not in ['int', 'float']
                         or self._profile[col].null_count > 0):
                     clean_samples.pop(col)
+                else:
+                    clean_column_ids.append(col_id)
         if len(clean_samples) <= 1:
             return None
 
         data = pd.DataFrame(clean_samples)
         data = data.apply(pd.to_numeric, errors='coerce')
-        return data.corr()
+
+        n_cols = len(self._profile)
+        corr_mat = np.empty((n_cols, n_cols))
+        corr_mat.fill(np.nan)
```

Comment:
`np.full((n_cols, n_cols), np.nan)`

Reply:
`fill` is actually faster: https://stackoverflow.com/questions/1704823/create-numpy-matrix-filled-with-nans

Reply:
I'm not sure if speed is what is needed here as much as cleanliness of code as `n` should be relatively small.  I'll leave the choice up to you.

Reply:
i've done some tests, it's not so much difference when n is large, so I changed to np.full for readability